tests/files.cpp

```diff
@@ -142,6 +146,15 @@ int main()
   FILE *n = fopen("/dev/null", "w");
   printf("5 bytes to dev/null: %zu\n", fwrite(data, 1, 5, n));
   fclose(n);
+  
+  // Test file creation with O_TRUNC (regression test for #16784)
+  const char *target_file = "test.out";
+  int target_fd;
+  target_fd = open(target_file, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR);
```

Comment:
Just one line should be fine here:

`int target_fd = open("test.out", O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR);`

Otherwise looks good.

Reply:
In fact maybe just call this `fd`.. I'm not sure `target_fd` has a meaning does it?